parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -384,40 +384,96 @@ should_skip_nvidia_drivers() {
     echo "$should_skip"
 }
 
-start_watch () {
-  capture_time=$(date +%s)
-  start_timestamp=$(date +%H:%M:%S)
+installJq () {
+  # jq is not available until downloaded in install-dependencies.sh with the installDeps function
+  # but it is needed earlier to call the capture_benchmarks function in pre-install-dependencies.sh
+  output=$(jq --version)
+  if [ -n "$output" ]; then
+    echo "$output"
+  else
+    if [[ $OS == $MARINER_OS_NAME ]]; then
+      sudo tdnf install -y jq && echo "jq was installed: $(jq --version)"
+    else
+      apt_get_install 5 1 60 jq && echo "jq was installed: $(jq --version)"
+    fi
+  fi
 }
 
-stop_watch () {
+capture_benchmark () {
+
+  set +x
+  # add current section title to the benchmarks array as a string
+  benchmarks+=($1)
+  # use nameref variable to hold that section title to reference that specific section array later in the function
+  declare -n current_section="${benchmarks[-1]}"
```

Comment:
Just a note FYR: I also used array index [-1] but in the E2E or test build VHD (I forgot which) it complains -1 is invalid, although my local unit test didn't complain about it. Then I found that this is a new syntax only supported in newer version of bash. If you run into index error later, this will be the place.

Reply:
Thank you. What syntax did you end up using? It has worked just fine in the test pipeline so far

Reply:
I needed to get the last element by getting the size of the array `#{array[@]}` and minus 1. Mind that if array size =0, you will still need to handle the potential -1.